docs/cpp_build_systems.md

```diff
@@ -0,0 +1,350 @@
+# How Protobuf supports multiple C++ build systems
+
+This document explains how the Protobuf project supports multiple C++ build
+systems.
+
+## Background
+
+Protobuf uses [Bazel](https://bazel.build) as the source of truth for building
+the Protobuf C++ runtime and Protobuf compiler[^historical_sot]. However, there
+are several different build systems in common use for C++, each one of which
+requires essentially a complete copy of build definitions.
+
+[^historical_sot]:
+  On a historical note, prior to its [release as Open Source
+  Software](https://opensource.googleblog.com/2008/07/protocol-buffers-googles-data.html),
+  the Protobuf project was developed using Blaze, Google's internal build
+  system, and predecessor to Bazel (the vast majority of Google's contributions
+  continue to be developed this way). The Open Source Protobuf project, however,
+  historically used Autoconf as the source of truth for C++ build correctness.
+  Over time, other build systems (including Bazel) have been added, thanks in
+  large part to substantial contributions from the Open Source community. Since
+  the Protobuf project deals with multiple languages (all of which ultimately
+  rely upon C++, for the Protobuf compiler), Bazel is a natural choice for a
+  project-wide source of truth -- in fact, Bazel (and its predecessor, Blaze)
```

Comment:
source of truth --> build system

-- in fact --> . In fact

Reply:
Done.